CMakeLists.txt

```diff
@@ -116,10 +116,54 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_STANDARD 17)
 if(MSVC)
-    string(APPEND CMAKE_CXX_FLAGS " /EHsc")
-    if(CMAKE_BUILD_TYPE STREQUAL "Release")
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /Zi /guard:cf")
-        set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /DEBUG /debugtype:cv,fixup /guard:cf")
+    # either MSVC, or clang-cl
```

Comment:
In general, we should either:
1. `add_compile_options()` / `add_link_options()`
2. Move these to a Toolchain file

Modifying CMAKE_C(XX)_FLAGS in the CMakeLists.txt is always a smell. That said, this works and follows what came before, so not requiring changes at this time.